formal-models/README.md

```diff
@@ -0,0 +1,135 @@
+   The node may become "dead" at any step in the middle of the consensus process.
+   Once the node becomes "dead" there's no way for it to rejoin the consensus
+   process.
+2. "Faulty" nodes. "Faulty" node is allowed to send consensus messages of *any*
+   type at *any* step and to change its view without regarding the dBFT view
+   changing rules. The node may become "faulty" at any step in the middle of the
+   consensus process. Once the node becomes "faulty" there's no way for it to
+   become "good" again.
+
+The specification contains several invariants and liveness properties that must
+be checked by the TLC Model Checker. These formulas mostly describe two basic
+concepts that dBFT algorithm expected to guarantee:
+
+1. No fork must happen. There must be no situation such that two different
+   blocks are accepted at two different consensus rounds (views).
+2. The block must always be accepted. There must be no situation such that nodes
+   are stuck in the middle of consensus process and can't take any further steps.
+
+The specification is written and working under several assumptions:
+
+1. All consensus messages are valid. In real life it is guaranteed by verifiable
+   message signatures. In case if malicious or corrupted message is received it
+   won't be handled by the node.
+2. The exact timeouts (e.g. t/o on waiting a particular consensus message, etc.)
+   are not included into the model. However, the model covers timeouts in
+   general, i.e. the timeout is just the possibility to perform a particular
+   state transition.
+3. The maximum number of consensus rounds (views) is restricted. This constraint
+   was introduced to reduce the number of possible model states to be checked.
+   The threshold may be specified via model configuration, and it is highly
+   recommended to keep this setting less or equal to the number of consensus
+   nodes.
+
+Here you can find the specification file and the TLC Model Checker launch
+configuration:
+
+* [TLA⁺ specification](./dbft/dbft.tla)
+* [TLC Model Checker configuration](./dbft/dbft___AllGoodModel.launch)
+
+### Extended dBFT 2.0 model
+
+This is an experimental dBFT 2.0 specification that extends the
+[basic model](#basic-dbft-20-model) in the following way: besides the shared pool
+of consensus messages `msgs` each consensus node has its own local pool of
+received and handled messages. Decisions on transmission between the node states
+are taken by the node based on the state of the local message pool. This approach
+allows to create more accurate low-leveled model which is extremely close to the
+dBFT implementation presented in this repository. At the same time such approach
+*significantly* increases the number of considered model states which leads to
+abnormally long TLC Model Checker runs. Thus, we do not recommend to use this
+model in development and place it here as an example of alternative (and more
+detailed) dBFT specification.
```

Comment:
Say specifically that in theory these models are equivalent?

Reply:
Fixed. Actually, TLA+ have a nice feature of *specification implementations* which perfectly suites for the case of Extended model, but I didn't do this.